### PR TITLE
fix: using and i18n work better

### PR DESCRIPTION
**Pull request by nlunets**
Fix for #834 and #833 hopefully

### CODE REVIEW

File: packages/fe-mockserver-plugin-cds/src/common.i18n.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Draft_DraftAdministrativeData: 'Draft Administrative Data',
+    Draft_DraftUUID: 'Draft (Technical ID)',
+    Draft_CreationDateTime: 'Draft Created On',
+    Draft_CreatedByUser: 'Draft Created By',
+    Draft_DraftIsCreatedByMe: 'Draft Created By Me',
+    Draft_LastChangeDateTime: 'Draft Last Changed On',
+    Draft_LastChangedByUser: 'Draft Last Changed By',
+    Draft_InProcessByUser: 'Draft In Process By',
+    Draft_DraftIsProcessedByMe: 'Draft In Process By Me'
+};
```
Comment (heimwege) [inline packages/fe-mockserver-plugin-cds/src/common.i18n.ts:31]: `as const`?

Comment (nlunets) [reply packages/fe-mockserver-plugin-cds/src/common.i18n.ts]: done

File: packages/fe-mockserver-plugin-cds/src/index.ts [hunk 2]
```diff
@@ -26,6 +27,29 @@ export default class CDSMetadataProvider implements IMetadataProcessor {
      */
     constructor(private fileLoader: IFileLoader, private options?: CDSMetadataProviderOptions) {}
 
+    async loadI18nMapFolder(targetFolder: string, i18nMap: Record<string, string>): Promise<void> {
+        if (await this.fileLoader.exists(targetFolder)) {
+            const i18nProp = await this.fileLoader.loadFile(targetFolder + 'i18n.properties');
+            const i18nPropLines = i18nProp.split('\n');
```
Comment (drktfl) [inline packages/fe-mockserver-plugin-cds/src/index.ts:33]: leads to somewhat different results if the .properties file uses CRLF line endings... (the value keeps the `\r`) - did not check if that is a problem / whether trimming would be an option?

File: packages/fe-mockserver-plugin-cds/src/index.ts [hunk 3]
```diff
@@ -26,6 +27,29 @@ export default class CDSMetadataProvider implements IMetadataProcessor {
...
+            for (const line of i18nPropLines) {
+                const [key, value] = line.split('=');
```
Comment (drktfl) [inline packages/fe-mockserver-plugin-cds/src/index.ts:35]: Not 100% sure, so better leave it as it is now: I think something like this edge case
```
key=This does not work if X=5
```
is valid in terms of .properties files, but would not work here.

Comment (heimwege) [reply packages/fe-mockserver-plugin-cds/src/index.ts:35]: You can use a regex in the `split`. Something like `const [key, value] = line.split(/=(.*)/s);` should do. The array might contain more than two entries but the others are empty and can be ignored.

File: packages/fe-mockserver-plugin-cds/src/index.ts (hunk 2 above)
Comment (nlunets) [reply packages/fe-mockserver-plugin-cds/src/index.ts:33]: trimming would help i guess :)

File: packages/fe-mockserver-plugin-cds/src/index.ts (hunk 3 above)
Comment (nlunets) [reply packages/fe-mockserver-plugin-cds/src/index.ts:35]: done

**Review by drktfl — APPROVED** [review]
looks ok for me :-)

**Pull request merged**